prbt_hardware_support/test/include/prbt_hardware_support/service_client_mock.h

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2019 Pilz GmbH & Co. KG
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef PRBT_HARDWARE_SUPPORT_SERVICE_CLIENT_MOCK_H
+#define PRBT_HARDWARE_SUPPORT_SERVICE_CLIENT_MOCK_H
+
+#include <functional>
+#include <string>
+
+#include <gmock/gmock.h>
+
+namespace prbt_hardware_support_tests
+{
+
+/**
+ * @brief Mock for a ros::ServiceClient, calls are passed through to a given callback (which can be a mock method).
+ *
+ * @tparam S service type
+ */
+template <typename S>
+class ServiceClientMock
```

Comment:
Is the ServiceClientMock class useful in more general and could be placed inside test_utils?

Reply:
yes!